pkg/controller/rmdworkload/rmdworkload_controller_test.go

```diff
@@ -470,6 +495,1169 @@ func TestRmdWorkloadControllerReconcile(t *testing.T) {
+					Name:      "rmd-workload-2",
+					Namespace: "default",
+				},
+				Spec: intelv1alpha1.RmdWorkloadSpec{
+					Nodes: []string{"example-node.com"},
+				},
+			},
+			rmdPods: &corev1.PodList{
+				Items: []corev1.Pod{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "rmd-example-node.com",
+							Namespace: "default",
+						},
+						Spec: corev1.PodSpec{
+							Containers: []corev1.Container{
+								{
+									Ports: []corev1.ContainerPort{
+										{
+											ContainerPort: 8080,
+										},
+									},
+								},
+							},
+						},
+						Status: corev1.PodStatus{
+							PodIPs: []corev1.PodIP{
+								{
+									IP: "127.0.0.1",
+								},
+							},
+						},
+					},
+				},
+			},
+			getWorkloadsResponse: map[string][]rmdtypes.RDTWorkLoad{
+				"127.0.0.1:8080": {
+					{
+						UUID: "rmd-workload-1",
+						ID:   "1",
+					},
+				},
+			},
+			expectedWorkloads: []targetedNodeInfo{
+				{
+					nodeName:       "example-node.com",
+					rmdAddress:     "http://127.0.0.1:8080",
+					workloadExists: false,
+				},
+			},
+			expectedErr: false,
+		},
+		{
+			name: "test case 2 - workload to be updated",
+			request: reconcile.Request{
+				types.NamespacedName{
+					Namespace: "default",
+					Name:      "rmd-workload-2",
+				},
+			},
+			rmdNodeStates: &intelv1alpha1.RmdNodeStateList{
+				Items: []intelv1alpha1.RmdNodeState{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "rmd-node-state-example-node.com",
+							Namespace: "default",
+						},
+						Spec: intelv1alpha1.RmdNodeStateSpec{
+							Node: "example-node.com",
+						},
+					},
+				},
+			},
+			rmdWorkload: &intelv1alpha1.RmdWorkload{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "rmd-workload-2",
+					Namespace: "default",
+				},
+				Spec: intelv1alpha1.RmdWorkloadSpec{
+					Policy: "gold",
+					Nodes:  []string{"example-node.com"},
+				},
+			},
+			rmdPods: &corev1.PodList{
+				Items: []corev1.Pod{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "rmd-example-node.com",
+							Namespace: "default",
+						},
+						Spec: corev1.PodSpec{
+							Containers: []corev1.Container{
+								{
+									Ports: []corev1.ContainerPort{
+										{
+											ContainerPort: 8080,
+										},
+									},
+								},
+							},
+						},
+						Status: corev1.PodStatus{
+							PodIPs: []corev1.PodIP{
+								{
+									IP: "127.0.0.1",
+								},
+							},
+						},
+					},
+				},
+			},
+			getWorkloadsResponse: map[string][]rmdtypes.RDTWorkLoad{
+				"127.0.0.1:8080": {
+					{
+						UUID: "rmd-workload-1",
+						ID:   "1",
+					},
+					{
+						UUID: "rmd-workload-2",
+						ID:   "2",
+					},
+				},
+			},
+			expectedWorkloads: []targetedNodeInfo{
+				{
+					nodeName:       "example-node.com",
+					rmdAddress:     "http://127.0.0.1:8080",
+					workloadExists: true,
+				},
+			},
+			expectedErr: false,
+		},
+		{
+			name: "test case 3 - pod IP missing",
```

Comment:
I think we can omit this test case.

It is causing a failure - so we _could_ patch the code itself to handle it, but the scenario itself is so unlikely that it's not really necessary.